File: src/main/java/com/pan/student/Student.java
package com.pan.student;
// 快捷鍵help / keymap
public class Student {
    //類別下的變數為屬性
    //預設字串為null
    //預設數字為0
    String id;
    String name;
    int english;
    int math;
    // java生成預設的建構子
    public Student() {
        super();
    }
    //tab selected param
    public Student(/*field*/String name, int english, int math) {
//        將區域變數傳進變成屬性(紫色)
        this.name = name;
        this.english = english;
        this.math = math;

    }
    //alt inser

    public void print() {
        System.out.println(name +  "\t english score = " + english + "\t math score = " + math);
    }

//    16
//    成績回傳
    public int hightest() {
        /*
        int max = (english > math) ? english : math; //三元運算式

        if (english > math) {
            max = english;
        }else {
            max = math;
        }
        */
        return (english > math) ? english : math;
    }
}

File: src/main/kotlin/com/pan/student/Student.kt
package com.pan.student

import java.util.*

//top -level
fun main(args: Array<String>) {
//    in在kotlin中是關鍵字，加上重音符號即可使用關鍵字
    val scanner = Scanner(System.`in`)
    print("Please enter student's name: ")
    val name = scanner.next()
    print("Please enter student's english score: ")
    val english = scanner.nextInt()
    print("Please enter student's math score: ")
    val math = scanner.nextInt()
    val stu = Students( name, english , math )
    stu.print()

}

class Students(/*property*/private var name:String, private var english: Int, private var math: Int) {
//    新增屬性，可跟建構子同時設計
    fun print() {
        println(name + "\t" + english + "\t" + math +
                "\t" + ( english + math ) / 2)
    }
}